cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java

```diff
@@ -105,7 +109,9 @@ public void addModule(final String name, final String className, final Location
         @Override
         public Void call(MDSDatasets datasets) throws DatasetModuleConflictException {
           DatasetModuleMeta existing = datasets.getTypeMDS().getModule(name);
-          if (existing != null) {
+          boolean allowForceDatasetUpgrade =
```

Comment:
How about caching it in the class's field? This applies everywhere - it would reduce amount of changes a lot I'd assume